src/ripple/app/tx/impl/AMMBid.cpp

```diff
@@ -173,8 +173,16 @@ applyBid(
         return {tecINTERNAL, false};
     STAmount const lptAMMBalance = (*ammSle)[sfLPTokenBalance];
     auto const lpTokens = ammLPHolds(sb, *ammSle, account_, ctx_.journal);
+    // should not happen
+    assert(
+        !ctx_.view().rules().enabled(fixInnerObjTemplate) ||
+        ammSle->isFieldPresent(sfAuctionSlot));
     if (!ammSle->isFieldPresent(sfAuctionSlot))
```

Comment:
This needs to be protected by an an amendment check. In general, when adding transaction breaking code, the code should follow the pattern:
```
if (amendment_on){
// do the new thing
}
else{
// do the old thing
}
```
It makes the code much easier to audit and makes it much less likely that transaction breaking changes will sneak in.